Microsoft.Toolkit.Graph.Providers.Uwp/WindowsProvider.cs

```diff
@@ -0,0 +1,304 @@
+        }
+
+        /// <summary>
+        /// Attempts to sign in the logged in user automatically.
+        /// </summary>
+        /// <returns>Success boolean.</returns>
+        public async Task<bool> TrySilentSignInAsync()
+        {
+            if (State == ProviderState.SignedIn)
+            {
+                return false;
+            }
+
+            State = ProviderState.Loading;
+
+            var tokenCredential = GetCredentialFromLocker();
+            if (tokenCredential == null)
+            {
+                // There is no credential stored in the locker.
+                State = ProviderState.SignedOut;
+                return false;
+            }
+
+            // Populate the password (aka token).
+            tokenCredential.RetrievePassword();
+
+            // Log the user in by storing the credential in memory.
+            _currentUser = new AuthenticatedUser(tokenCredential);
+
+            try
+            {
+                var me = await Graph.Me.Request().GetAsync();
+
+                // Update the state to be signed in.
+                State = ProviderState.SignedIn;
+                return true;
+            }
+            catch
+            {
+                // Update the state to be signed in.
+                State = ProviderState.SignedOut;
+                return false;
+            }
+        }
+
+        /// <inheritdoc />
+        public override Task LoginAsync()
+        {
+            if (State == ProviderState.SignedIn)
+            {
+                return Task.CompletedTask;
+            }
+
+            State = ProviderState.Loading;
+
+            if (_currentPane != null)
+            {
+                _currentPane.AccountCommandsRequested -= BuildPaneAsync;
+            }
+
+            _currentPane = AccountsSettingsPane.GetForCurrentView();
+            _currentPane.AccountCommandsRequested += BuildPaneAsync;
+
+            AccountsSettingsPane.Show();
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public override Task LogoutAsync()
+        {
+            if (State == ProviderState.SignedOut)
+            {
+                return Task.CompletedTask;
+            }
+
+            State = ProviderState.Loading;
+
+            if (_currentPane != null)
+            {
+                _currentPane.AccountCommandsRequested -= BuildPaneAsync;
+                _currentPane = null;
+            }
+
+            if (_currentUser != null)
+            {
+                // Remove the user info from the PaasswordVault
+                var vault = new Windows.Security.Credentials.PasswordVault();
+                vault.Remove(_currentUser?.TokenCredential);
+
+                _currentUser = null;
+            }
+
+            State = ProviderState.SignedOut;
+            return Task.CompletedTask;
+        }
+
+        /// <inheritdoc />
+        public override Task AuthenticateRequestAsync(HttpRequestMessage request)
```

Comment:
I think the root/crux of our issues is that this method is required from `IAuthenticationProvider` which we only get from pulling in the whole Graph SDK (well auth package currently, but eventually), eh?

That means even if we wanted to use the Windows Provider without Graph we're still pulling in Graph? Like what other scenarios can we use the provider for outside of graph? (Or I guess if you decided to use this to add the header to general rest call still without the graph sdk explicitly?)

@shweaver-MSFT @nmetulev I think this is the big issue we need to understand how to make everything work now and in the future together with @darrelmiller.